src/libraries/System.Private.CoreLib/src/System/Globalization/EastAsianLunisolarCalendar.cs

```diff
@@ -646,7 +646,7 @@ public override bool IsLeapYear(int year, int era)
             return InternalIsLeapYear(year);
         }
 
-        private const int DefaultGregorianTwoDigitYearMax = 2029;
+        private const int DefaultGregorianTwoDigitYearMax = 2049;
```

Comment:
incidentally, I am not sure why this
https://github.com/dotnet/runtime/blob/43ebf1a159c0af6ea8c4364d1e9d881972561914/src/libraries/System.Private.CoreLib/src/System/Globalization/EastAsianLunisolarCalendar.cs#L657
and this are different
https://github.com/dotnet/runtime/blob/43ebf1a159c0af6ea8c4364d1e9d881972561914/src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendar.cs#L417

Reply:
In the EastAsianLunisolarCalendar file, I've made the call to GetSystemTwoDigitYearSetting match how it's handled in GregorianCalendar. Let me know if it should be the other way around

Reply:
I don't see that change.

Reply:
> incidentally, I am not sure why this

`GetYear` maps the Gregorian year to the calendar year. I don't think we need to change that.

Reply:
> In the EastAsianLunisolarCalendar file, I've made the call to GetSystemTwoDigitYearSetting match how it's handled in GregorianCalendar. Let me know if it should be the other way around

EastAsianLunisolarCalendar is abstract class which any user can subclass and introduce a new calendar. You never know what the calendar does. So we need to avoid any breaking with that.

Reply:
This change should now be reverted to the original code

Reply:
why?

Reply:
Sorry, I meant I have reverted the change to the `GetSystemTwoDigitYearSetting` call to the original code per your request

Reply:
Got it :-) thanks!